File: Contribute/Plugin Samples/VB.NET/XMLParserPlugin/3rd Party Control/IXmlControl.vb
Imports System.Drawing.Printing

Namespace XMLTreeView
	''' <summary>
	''' Summary description for IXmlControl.
	''' </summary>
	Friend Interface IXmlControl
		#Region "Content"
		Property Xml() As String
		#End Region

		#Region "Searching"
		Sub Search(sender As Object, e As EventArgs)
		Sub StartSearch(criterion As String, caseSensitive As Boolean)
		Sub [Next]()
		#End Region

		#Region "Printing"
		Sub Print(sender As Object, e As EventArgs)
		Sub PrintPage(sender As Object, e As PrintPageEventArgs)
		ReadOnly Property PrintDocument() As PrintDocument
		#End Region

		#Region "Editing, Deleting, Copying, Pasting"
		Sub Edit(sender As Object, e As System.EventArgs)
		Sub Delete(sender As Object, e As System.EventArgs)
		Sub Copy(sender As Object, e As System.EventArgs)
		Sub Paste(sender As Object, e As System.EventArgs)
		#End Region

		#Region "Saving, Opening"
		Sub Save(sender As Object, e As System.EventArgs)
		#End Region
	End Interface
End Namespace

File: Contribute/Plugin Samples/C#/Parser/XMLParserPlugin/3rd Party Control/IXmlControl.cs
using System;
using System.Drawing.Printing;

namespace XMLTreeView
{
	/// <summary>
	/// Summary description for IXmlControl.
	/// </summary>
	internal interface IXmlControl
	{
		#region Content
		string Xml { get; set; }
		#endregion // Content

		#region Searching
		void Search( object sender, EventArgs e );
		void StartSearch( string criterion, bool caseSensitive );
		void Next();
		#endregion // Searching

		#region Printing
		void Print( object sender, EventArgs e );
		void PrintPage( object sender, PrintPageEventArgs e );
		PrintDocument PrintDocument { get; }
		#endregion // Printing

		#region Editing, Deleting, Copying, Pasting
		void Edit(object sender, System.EventArgs e);
		void Delete(object sender, System.EventArgs e);
		void Copy(object sender, System.EventArgs e);
		void Paste(object sender, System.EventArgs e);
		#endregion // Edititing, Deleting, Copying, Pasting

		#region Saving, Opening
		void Save(object sender, System.EventArgs e);
		#endregion // Saving, Opening
	}
}
